From 9f0c1849d4b81fae1c382791ae004aa63c92af6e Mon Sep 17 00:00:00 2001
From: Yasufumi Ogawa <ogawa.yasufumi@lab.ntt.co.jp>
Date: Sat, 24 Nov 2018 02:49:30 +0900
Subject: [PATCH 2/3] makefile: introduce GNUmakefile

As in DPDK, change Makefile to use gmake and add GNUmakefile.

Signed-off-by: Yasufumi Ogawa <ogawa.yasufumi@lab.ntt.co.jp>
---
 GNUmakefile | 38 ++++++++++++++++++++++++++++++++++++++
 Makefile    | 36 +-----------------------------------
 2 files changed, 39 insertions(+), 35 deletions(-)
 create mode 100644 GNUmakefile

diff --git a/GNUmakefile b/GNUmakefile
new file mode 100644
index 0000000..1096c47
--- /dev/null
+++ b/GNUmakefile
@@ -0,0 +1,38 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright(c) 2018 Nippon Telegraph and Telephone Corporation
+
+VERSION = 18.05.1
+
+ifeq ($(RTE_SDK),)
+$(error "Please define RTE_SDK environment variable")
+endif
+
+# Default target, can be overriden by command line or environment
+RTE_TARGET ?= x86_64-native-linuxapp-gcc
+
+include $(RTE_SDK)/mk/rte.vars.mk
+
+DIRS-y += src
+
+include $(RTE_SDK)/mk/rte.extsubdir.mk
+
+DOC_ROOT = docs/guides
+
+# Compile RST documents
+.PHONY: doc
+doc: doc-pdf doc-html
+
+.PHONY: doc-html
+doc-html:
+	make -C $(DOC_ROOT) html
+
+.PHONY: doc-pdf
+doc-pdf:
+	python $(DOC_ROOT)/gen_pdf_imgs.py
+	make -C $(DOC_ROOT) latexpdf
+	@echo "Succeeded to generate '$(DOC_ROOT)/_build/latex/SoftPatchPanel.pdf'"
+
+.PHONY: doc-clean
+doc-clean:
+	find $(DOC_ROOT)/images/ -type f -name "*.pdf" -delete
+	make -C $(DOC_ROOT) clean
diff --git a/Makefile b/Makefile
index d5423ca..964379f 100644
--- a/Makefile
+++ b/Makefile
@@ -29,38 +29,4 @@
 #   (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 #   OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-VERSION = 18.05.1
-
-ifeq ($(RTE_SDK),)
-$(error "Please define RTE_SDK environment variable")
-endif
-
-# Default target, can be overriden by command line or environment
-RTE_TARGET ?= x86_64-native-linuxapp-gcc
-
-include $(RTE_SDK)/mk/rte.vars.mk
-
-DIRS-y += src
-
-include $(RTE_SDK)/mk/rte.extsubdir.mk
-
-DOC_ROOT = docs/guides
-
-# Compile RST documents
-.PHONY: doc-pdf
-doc-all: doc-pdf doc-html
-
-.PHONY: doc-html
-doc-html:
-	make -C $(DOC_ROOT) html
-
-.PHONY: doc-pdf
-doc-pdf:
-	python $(DOC_ROOT)/gen_pdf_imgs.py
-	make -C $(DOC_ROOT) latexpdf
-	@echo "Succeeded to generate '$(DOC_ROOT)/_build/latex/SoftPatchPanel.pdf'"
-
-.PHONY: doc-clean
-doc-clean:
-	find $(DOC_ROOT)/images/ -type f -name "*.pdf" -delete
-	make -C $(DOC_ROOT) clean
+.error Error please compile using GNU Make (gmake)
-- 
2.7.4

